applications/crossbar/src/modules/cb_configs.erl

```diff
@@ -15,198 +12,164 @@
+save_diff(Context, ConfigId, Parent) ->
+    JObj = cb_context:doc(Context),
+    JObjDiff = kz_json:diff(JObj, Parent),
+    StoredDocument = kz_json:private_fields(kapps_account_config:get(cb_context:account_id(Context), ConfigId)),
+    Document = kz_json:merge_recursive(StoredDocument, JObjDiff),
+    flush(crossbar_doc:save(Context, Document, [])),
+    crossbar_doc:handle_datamgr_success(set_id(ConfigId, JObj), Context).
+
+-spec validate_diff(cb_context:context(), path_token(), kz_json:object()) -> cb_context:context().
+validate_diff(Context, ConfigId, Parent) ->
+    JObj = strip_id(cb_context:req_data(Context)),
+    FullConfig = kz_json:merge_recursive(Parent, JObj),
+    maybe_validate(ConfigId, FullConfig, Parent),
+    pass_validation(Context, FullConfig).
+
+-spec doc_id(ne_binary()) -> ne_binary().
+doc_id(ConfigId) -> kapps_account_config:config_doc_id(ConfigId).
+
+-spec set_id(ne_binary(), kz_json:object()) -> kz_json:object().
+set_id(ConfigId, JObj) -> kz_json:set_value(<<"id">>, doc_id(ConfigId), JObj).
+
+-spec strip_id(kz_json:object()) -> kz_json:object().
+strip_id(JObj) -> kz_json:delete_key(<<"id">>, JObj, prune).
+
+-spec pass_validation(cb_context:context(), kz_json:object()) -> cb_context:context().
+pass_validation(Context, JObj) ->
+    cb_context:setters(Context,[
+                                {fun cb_context:set_doc/2, JObj}
+                               ,{fun cb_context:set_resp_status/2, success}
+                               ]).
+
+-spec error_validation(cb_context:context()) -> cb_context:context().
+error_validation(Context) ->
+    cb_context:setters(Context, [
+                                 {fun cb_context:set_resp_error_code/2, 400}
+                                ,{fun cb_context:set_resp_status/2, error}
+                                ]).
+
+-spec maybe_validate(ne_binary(), kz_json:object(), kz_json:object()) -> skip_validation | valid.
+maybe_validate(ConfigId, ConfigIdObj, Parent) ->
+    SchemaName = kapps_config_util:account_schema_name(ConfigId),
+    case validate_schema(SchemaName, Parent) of
+        valid ->
+            valid = validate_schema(SchemaName, ConfigIdObj);
+        Error ->
+            ErrorMsg = parse_error(Error),
+            lager:error("Parent configuration for ~p doesn't pass schema ~p validation due to: ~p", [ConfigId, SchemaName, ErrorMsg]),
+            skip_validation
+    end.
+
+parse_error({invalid_document, [{data_invalid, _Schema, Error, _Doc, Path}]}) -> {Error, Path};
+parse_error(X) -> X.
+
+-spec validate_schema(api_binary(), kz_json:object()) -> no_schema_present | valid | {invalid_document, term()}.
+validate_schema(Name, JObj) ->
+    case kz_json_schema:load(Name) of
+        {ok, Schema} ->
+            case kz_json_schema:validate(Schema, JObj) of
+                {ok, _} -> valid;
+                {error, Error} -> {invalid_document, Error}
+            end;
+        _ -> no_schema_present
+    end.
```

Comment:
it doesn't allow to validate a json object, only context document, what is not the case here.